deps/wazuh_testing/wazuh_testing/tools/performance/statistic.py

```diff
@@ -40,44 +47,61 @@ class StatisticMonitor:
         dst_dir (str): directory to store the CSVs. Defaults to temp directory.
         csv_file (str): path to the CSV file.
         target (str): target file to monitor.
+        parse_json (bool): Determine if the file is a JSON file. Default False.
     """
 
-    def __init__(self, target='agent', time_step=5, dst_dir=gettempdir()):
+    def __init__(self, target='agent', time_step=5, dst_dir=gettempdir(), use_state_file=False, ip = 'localhost', port = '55000'):
         self.event = None
         self.thread = None
         self.time_step = time_step
         self.target = target
         self.dst_dir = dst_dir
+        self.use_state_file = use_state_file
+        self.ip = ip
+        self.port = port
+        self.daemon = None
         self.parse_json = False
 
+
         if self.target == 'agent':
             self.statistics_file = tls.AGENT_STATISTICS_FILE
+            self.use_state_file = True
         elif self.target == 'logcollector':
             self.statistics_file = tls.LOGCOLLECTOR_STATISTICS_FILE
+            self.use_state_file = True
             self.parse_json = True
         elif self.target == 'remote':
             self.statistics_file = tls.REMOTE_STATISTICS_FILE
+            self.daemon = 'wazuh-remoted'
         elif self.target == 'analysis':
             self.statistics_file = tls.ANALYSIS_STATISTICS_FILE
+            self.daemon = 'wazuh-analysisd'
+        elif self.target == 'wazuhdb':
+            self.daemon = 'wazuh-db'
         else:
             raise ValueError(f'The target {self.target} is not a valid one.')
 
-        state_file = splitext(basename(self.statistics_file))[0]
-        self.csv_file = join(self.dst_dir, f'{state_file}_stats.csv')
+        if self.use_state_file == True:
```

Comment:
```suggestion
        if self.use_state_file:
```

Reply:
Fixed in https://github.com/wazuh/wazuh-qa/pull/4970/commits/a1d0448b75f63ebcbe1d81489fae944108b17967